### ISSUE TITLE
weird yee shift for component in get_array_metadata

### ISSUE THREAD

**Issue by smartalecH** [NanoComp/meep#881]
This may be related to #838.

When trying to grab the coordinates of a single point in 3D coordinates (like a source), I sometimes get the error:
`weird yee shift for component`.

```python3
import meep as mp

cell = mp.Vector3(16,8,4)

geometry = [mp.Block(mp.Vector3(mp.inf,1,1),
                     center=mp.Vector3(),
                     material=mp.Medium(epsilon=12))]

sources = [mp.Source(mp.ContinuousSource(frequency=0.15),
                     component=mp.Ez,
                     center=mp.Vector3(-7,0,0))]

pml_layers = [mp.PML(1.0)]

resolution = 10

sim = mp.Simulation(cell_size=cell,
                    boundary_layers=pml_layers,
                    geometry=geometry,
                    sources=sources,
                    resolution=resolution)

sim.init_sim()
temp = sim.get_source(mp.Ez,center=sources[0].center,size=sources[0].size)
print(temp)

(x,y,z,w)=sim.get_array_metadata(center=sources[0].center,size=sources[0].size)
print(x)
print(y)
print(z)
```

Output:
```
Using MPI version 3.1, 1 processes
-----------
Initializing structure...
Working in 3D dimensions.
Computational cell is 16 x 8 x 4 with resolution 10
     block, center = (0,0,0)
          size (1e+20,1,1)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (12,12,12)
time for set_epsilon = 1.75841 s
-----------
0j
meep: weird yee shift for component none
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
```

The 2D results occasionally segfault. When the routine succeeds, it outputs:
```
Using MPI version 3.1, 1 processes
-----------
Initializing structure...
Working in 2D dimensions.
Computational cell is 16 x 8 x 0 with resolution 10
     block, center = (0,0,0)
          size (1e+20,1,1)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (12,12,12)
time for set_epsilon = 0.031008 s
-----------
0j
(-7.050000000000001, -6.950000000000001)
(-0.05, 0.05)
(0.0,)

Elapsed run time = 0.0366 s

Field time usage:
        outputting fields: 0.000228 s
          everything else: 0.001851 s```

**Comment by smartalecH** [comment]
It looks like this was due to a recent change. Older versions of MEEP don't seem to have this problem. We should add a test to prevent this in the future.

I temporarily work around this by commenting out the following line:
https://github.com/NanoComp/meep/blob/da6fb3a58d95aca39c1bff444f0fd11fbfc1f2c4/src/vec.cpp#L338

Obviously, this could have drastic consequences. But at least we know something seems to be going on with`stride()`.

**Comment by stevengj** [comment]
It looks like something is passing `NO_COMPONENT` when it should be passing `Centered` (`== Dielectric`).

**Comment by stevengj** [comment]
Should be fixed by 4f394234d0ca5df05b15ea9cee227d343211416d … if so, please close this.

**Comment by smartalecH** [comment]
The change now outputs the following:

```
(-7.050000000000001, -6.950000000000001)
(-0.05, 0.05)
(-0.05, 0.05)
```
looks good to me.
